capstone/src/main/java/com/google/sps/Objects/Post.java

```diff
@@ -1,21 +1,24 @@
 package com.google.sps.Objects;
 
 import java.util.ArrayList;
+import java.util.HashSet;
 
 public final class Post {
 
-  private final User author;
+  private final long postId;
+  private final String authorId;
```

Comment:
I think this is the same comment --- each user only has one profile picture. Each comment is only made by one user, but you don't want to have a bunch of copies of this user saved everywhere. If the DB handles this and there's ACTUALLY only one user saved, but it's referenced in multiple places then you could have a user saved here.

But from a user ID, you should be able to query the DB and get the user you need to get the profile picture or any other relevant data, right?

Reply:
Yes you should be able to query the DB with the userId but I think that would involve an extra step. If you return the userId with the comment, you would have to: get comment from DB -> return Comment object as JSON -> get user from DB using userId -> return User data as JSON.

Whereas if you stored the needed user data in the comment object, the process would look like this: get comment from DB -> use userId stored in DB to get needed user data -> return Comment object as JSON.

I'm not 100% sure if I'm thinking about this the right way but I don't think the first scenario would necessarily save space because you would still end up with multiple copies of the returned user data

Reply:
I think in the first scenario we wouldn't have to store the User object with the Comment in the database, we would just be storing a string for userId. Then we use the userId to get the relevant user information. So I think it does end up saving space in the database since we don't have to save all the User info with each Comment. It's just that now we need to take the userId and query the database each time to get the relevant User information

Reply:
Oh, I thought we would only be storing the userId in the Database to save space in both scenarios. The difference would be in the first scenario, we would return the userId as part of the Comment object while in the second scenario we would be returning a User/whatever user data we need as part of the Comment object.

Reply:
Okay, I see what you mean. The User data contains more data than we need, such as groups, interests, email and phone number. I think it would make sense to return a Comment Object with a all the User data if we want to be able to navigate to that user's profile and load all their info. But if we aren't able to navigate to their page then we only really need name and picture

Reply:
For a post, we need the userId, userImage, and userName. For a comment, we need the userId and userImage. As Andrew said above, if a user posted or commented 10 times, we wouldn't want 10 copies of that information. We could loop through all the userId's and just retrieve the user info once?

Reply:
Okay, that definitely makes sense